### DISCUSSION TITLE
Suggested task for Outreachy Dec 22: Task 1 - OSBv2 MDF notebook

### DISCUSSION THREAD

**Discussion by pgleeson** [ModECI/MDF#343] [Ideas]
A suggested set of tasks for applicants to the **Outreachy Winter 2022** call.

1) Create a new account on Open Source Brain v2: https://www.v2.opensourcebrain.org 
2) Create a new [workspace](https://docs.opensourcebrain.org/OSBv2/Workspaces.html) from the [ModECI MDF OSBv2 repository](https://www.v2.opensourcebrain.org/repositories/1)  (which points to the main MDF GitHub repository here: https://github.com/ModECI/MDF)
3) Open the workspace in the [JupyterLab application](https://docs.opensourcebrain.org/OSBv2/JupyterLab.html).
4) Install the latest version of MDF which has been copied into the workspace (launch a Terminal, go to `/opt/workspace/ModECI MDF/main`, type `pip install .`)
5) Open the simple example Jupyter notebook (at `/opt/workspace/ModECI MDF/main/examples/SimpleExample.ipynb`) and go through this to understand the main concepts in MDF. 
6) Create a new notebook in your workspace which creates an MDF model with 3 connected nodes, with a unique mathematical expression in each and at least one condition and/or a time derivative ([example](https://github.com/ModECI/MDF/blob/main/examples/MDF/States.yaml#L16)). It doesn't necessarily have to do anything real/useful (it's great if it does!) but just demonstrates your understanding of MDF. 
7) Share the workspace publicly on OSBv2
8) Email us a link to the workspace.  

To get a feel for what OSBv2 can be used for see the [Guided tour of OSBv2](https://docs.opensourcebrain.org/OSBv2/Guided_tour.html)

<img width="1680" alt="Screenshot 2022-10-10 at 11 14 55" src="https://user-images.githubusercontent.com/1556687/194847476-c39b8215-d71b-44e2-b504-2f6eb09cac25.png">

**Comment by kusanele** [comment]
Good day Family, Looking at the structure of this project. Do we still need to record our contribution??

**Reply by sanjayankur31** [reply]
Yes, please record all contributions. You'll be asked to note these in the application (and then these are what we mentors review to pick our interns)

**Reply by kusanele** [reply]
Noted, i just emailed you the link to my notebook. Kindly have a look. Thank you

**Comment by ManasviEmmadi** [comment]
Hello all! I have done this task for the summer outreachy'23.

**Comment by Jess-chege** [comment]
hello all I need help on how to go about here 
2.Create a new [workspace](https://docs.opensourcebrain.org/OSBv2/Workspaces.html) from the [ModECI MDF OSBv2 repository](https://www.v2.opensourcebrain.org/repositories/1) (which points to the main MDF GitHub repository here: https://github.com/ModECI/MDF)

**Reply by sanjayankur31** [reply]
Hi @Jess-chege : you don't have to use OSBv2 here. You can do the task on your local machine once you have your virtual environments set up.

**Reply by pgleeson** [reply]
We have changed our OSBv2 homepage recently, meaning unfortunately links like https://www.v2.opensourcebrain.org/repositories/1 are broken, but https://v2.opensourcebrain.org/repositories/1 works correctly. We will resolve this issue soon, but for now, try removing the www.

Note though we have not updated the version of MDF on OSB recently, so it is indeed better to run the code and make/test changes locally.
